internal/pkg/service/stream/storage/metacleanup/metacleanup.go

```
@@ -205,3 +245,72 @@ func (n *Node) isFileExpired(file model.File, age time.Duration) bool {
+	defer span.End(&err)
+
+	var keboolaJob keboolaSink.Job
+	// Retrieve job on bridge level, the mutex is not needed, we will check state of job through API
+	if keboolaJob, err = n.bridge.Job(job.JobKey).Do(ctx).ResultOrErr(); err != nil {
+		err = errors.PrefixErrorf(err, `cannot get keboola storage job "%s"`, job.JobKey)
+		n.logger.Error(ctx, err.Error())
+		return err, false
+	}
+
+	// Get job details from storage API
+	api := n.publicAPI.NewAuthorizedAPI(keboolaJob.Token, 1*time.Minute)
+	var jobStatus *keboola.StorageJob
+	if jobStatus, err = api.GetStorageJobRequest(keboolaJob.StorageJobKey).Send(ctx); err != nil {
+		n.logger.Warnf(ctx, "cannot get information about storage job, probably already deleted: %s", err.Error())
+		return nil, false
+	}
+
+	attrs = append(attrs, attribute.String("job.state", jobStatus.Status))
+	ctx = ctxattr.ContextWith(ctx, attrs...)
+	// Check status of storage Job
+	if jobStatus.Status == keboola.StorageJobStatusProcessing || jobStatus.Status == keboola.StorageJobStatusWaiting {
+		n.logger.Debugf(ctx, "cannot remove storage job, job status: %s", jobStatus.Status)
+		return nil, false
+	}
+
+	// Acquire lock
+	mutex := n.locks.NewMutex(fmt.Sprintf("api.source.sink.jobs.%s", job.SinkKey))
+	if err = mutex.TryLock(ctx); err != nil {
+		return err, false
+	}
+	defer func() {
+		if err := mutex.Unlock(ctx); err != nil {
+			n.logger.Errorf(ctx, "cannot unlock the lock: %s", err)
+		}
+	}()
+
+	// Delete job on bridge level
+	if _, err = n.bridge.DeleteJob(job.JobKey).RequireLock(mutex).Do(ctx).ResultOrErr(); err != nil {
+		err = errors.PrefixErrorf(err, `cannot delete finished storage job "%s"`, job.JobKey)
+		n.logger.Error(ctx, err.Error())
+		return err, false
+	}
+
+	// Purge the job
+	if err = n.storageRepository.Job().Purge(&job).RequireLock(mutex).Do(ctx).Err(); err != nil {
+		err = errors.PrefixErrorf(err, `cannot delete finished job "%s"`, job.JobKey)
+		n.logger.Error(ctx, err.Error())
+		return err, false
+	}
```

This will cause error when we soft delete `sink` and we do not delete `storage/keboola/jobs`. Should we suppress the error or leave it fail? @jachym-tousek-keboola